### ISSUE TITLE
ref sugar watch has a type error

### ISSUE THREAD

**Issue by nshusr** [johnsoncodehk/vue-tsc#18]
``` js
ref: selectTable = 'record'

watch(
  $selectTable,
  select => {
    router.push({
      query: {
        selectTable: select,
      },
    })
  },
  {
    immediate: true,
  }
)
```

build
```
The last overload gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'object'.

     $selectTable,

@vue+runtime-core@3.0.11/node_modules/@vue/runtime-core/dist/runtime-core.d.ts:1394:25
    1394 export declare function watch<T extends object, Immediate extends Readonly<boolean> = false>(source: T, cb: WatchCallback<T, Immediate extends true ? (T | undefined) : T>, options?: WatchOptions<Immediate>): WatchStopHandle;

```

**Comment by johnsoncodehk** [comment]
Ref sugar is not support yet, will fix in next version, thanks for your report!

**Comment by johnsoncodehk** [comment]
Fixed in 0.0.25.
